index.js

```diff
@@ -45,9 +44,488 @@ app.intent('Default Welcome Intent', conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  let { convs } = conv.contexts.input['addparticipantgroup_data'].parameters;
+  let user = conv.parameters.user;
+  let users = await searchUsers(circuit, conv, user);
+
+  if (users.length !== 1) {
+    // Stays in this intent and gets input from user again
+    return;
+  }
+
+  // Save user to context because only one found
+  conv.contexts.set('addparticipantgroup_data', 5, { users: users });
+  conv.contexts.delete('addparticipantgroup_getuser');
+
+  if (!convs.length) {
+    // No conversation found
+    conv.ask(`Thank you. I did not find the conversation name that you gave me earlier. What is it again?`);
+    conv.contexts.set('addparticipantgroup_getconv', 5);
+    return;
+  } else if (convs.length > 1) {
+    // Multiple conversations found.
+    convs = convs.slice(0, Math.min(7, convs.length));
+    const suggestions = convs.map(c => c.topic);
+
+    conv.ask(`Thank you. I found more than one result for the conversation name you gave me earlier. What's the full name?`, new Suggestions(suggestions));
+    conv.contexts.set('addparticipantgroup_getconv', 5);
+    return;
+  }
+
+  // One result found for user and conversation
+  conv.ask(`Ready to add ${user} to ${convs[0].topic}?`, new Suggestions('Yes', 'No'));
+});
+
+/**
+ * add.participant.to.group - no
+ */
+app.intent('add.participant.to.group - no', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  conv.ask('Is there anything else I can do for you?', new Suggestions('Yes', 'No'));
+  conv.contexts.delete('addparticipantgroup_data');
+  conv.contexts.set('anything_else', 2);
+});
+
+/**
+ * add.participant.to.group - yes
+ */
+app.intent('add.participant.to.group - yes', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  const { users, convs } = conv.contexts.input['addparticipantgroup_data'].parameters;
+  const thisUser = circuit.user;
+
+  if (convs[0].participants.indexOf(users[0].userId) !== -1) {
+    // User is already in the conversation
+    conv.ask(`${users[0].displayName} is already a participant in ${convs[0].topic}. Is there anything else I can do for you?`, new Suggestions('Yes', 'No'));
+    conv.contexts.delete('addparticipantgroup_data');
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  if (convs[0].isModerated && convs[0].moderators.indexOf(thisUser.userId) === -1) {
+    // Conversation is moderated and user is not a moderator
+    conv.ask(`Sorry, but you are not a moderator in ${convs[0].topic} so ${users[0].displayName} cannot be added. Is there anything else I can do for you?`, new Suggestions('Yes', 'No'));
+    conv.contexts.delete('addparticipantgroup_data');
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  await circuit.addParticipant(convs[0].convId, users[0].userId, true);
```

Comment:
I'm not sure if you want to addParticipant to call, I might leave out the third param to default it to not adding them into a call. This depends on the sue case.